packages/pricing/src/services/price-set-money-amount-rules.ts

```diff
@@ -0,0 +1,124 @@
+import { Context, DAL, FindConfig, PricingTypes } from "@medusajs/types"
+import {
+  InjectManager,
+  InjectTransactionManager,
+  MedusaContext,
+  ModulesSdkUtils,
+  retrieveEntity,
+} from "@medusajs/utils"
+import { PriceSetMoneyAmountRules } from "@models"
+
+import { doNotForceTransaction, shouldForceTransaction } from "@medusajs/utils"
+
+type InjectedDependencies = {
+  priceSetMoneyAmountRulesRepository: DAL.RepositoryService
+}
+
+export default class PriceSetMoneyAmountRulesService<
+  TEntity extends PriceSetMoneyAmountRules = PriceSetMoneyAmountRules
+> {
+  protected readonly priceSetMoneyAmountRulesRepository_: DAL.RepositoryService
+
+  constructor({ priceSetMoneyAmountRulesRepository }: InjectedDependencies) {
+    this.priceSetMoneyAmountRulesRepository_ =
+      priceSetMoneyAmountRulesRepository
+  }
+
+  @InjectManager("priceSetMoneyAmountRulesRepository_")
+  async retrieve(
+    priceSetMoneyAmountRulesId: string,
+    config: FindConfig<PricingTypes.PriceSetMoneyAmountRulesDTO> = {},
+    @MedusaContext() sharedContext: Context = {}
+  ): Promise<TEntity> {
+    return (await retrieveEntity<
+      PriceSetMoneyAmountRules,
+      PricingTypes.PriceSetMoneyAmountRulesDTO
+    >({
+      id: priceSetMoneyAmountRulesId,
+      identifierColumn: "id",
+      entityName: PriceSetMoneyAmountRules.name,
+      repository: this.priceSetMoneyAmountRulesRepository_,
+      config,
+      sharedContext,
+    })) as TEntity
+  }
+
+  @InjectManager("priceSetMoneyAmountRulesRepository_")
+  async list(
+    filters: PricingTypes.FilterablePriceSetMoneyAmountRulesProps = {},
+    config: FindConfig<PricingTypes.PriceSetMoneyAmountRulesDTO> = {},
+    @MedusaContext() sharedContext: Context = {}
+  ): Promise<TEntity[]> {
+    return (await this.priceSetMoneyAmountRulesRepository_.find(
+      this.buildQueryForList(filters, config),
+      sharedContext
+    )) as TEntity[]
+  }
+
+  @InjectManager("priceSetMoneyAmountRulesRepository_")
+  async listAndCount(
+    filters: PricingTypes.FilterablePriceSetMoneyAmountRulesProps = {},
+    config: FindConfig<PricingTypes.PriceSetMoneyAmountRulesDTO> = {},
+    @MedusaContext() sharedContext: Context = {}
+  ): Promise<[TEntity[], number]> {
+    return (await this.priceSetMoneyAmountRulesRepository_.findAndCount(
+      this.buildQueryForList(filters, config),
+      sharedContext
+    )) as [TEntity[], number]
+  }
+
+  private buildQueryForList(
+    filters: PricingTypes.FilterablePriceSetMoneyAmountRulesProps = {},
+    config: FindConfig<PricingTypes.PriceSetMoneyAmountRulesDTO> = {}
+  ) {
+    const queryOptions = ModulesSdkUtils.buildQuery<PriceSetMoneyAmountRules>(
+      filters,
+      config
+    )
+
+    if (filters.id) {
```

Comment:
question: The build query already transform to $in if the value is an array, any reason why we are doing that here?

Reply:
I thought so too, if its not needed, will remove it from everywhere.